From abf72613b397f5d88d6bc294fd796a3108d2a3a7 Mon Sep 17 00:00:00 2001
From: Peng Fan <peng.fan@nxp.com>
Date: Thu, 22 Mar 2018 11:06:36 +0800
Subject: [PATCH 3522/5242] MLK-17788 soc: imx: ipc: not abort when set wake
 error

commit  89a61aa02356177bb83141bed87e8ca050c362fb from
https://source.codeaurora.org/external/imx/linux-imx.git

When irq_set_irq_wake error, it means no irq wakeup capability,
showing the error msg is enough, no need to abort and cause
kernel stop.

For xen, currently we do not support suspend/resume, and
no wu interrupt controller support now, so need to remove
"return err" to avoid kernel stop.

Signed-off-by: Peng Fan <peng.fan@nxp.com>
Reviewed-by: Anson Huang <Anson.Huang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/soc/imx/sc/main/ipc.c |    6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/drivers/soc/imx/sc/main/ipc.c b/drivers/soc/imx/sc/main/ipc.c
index 55516d2..5a1fa02 100644
--- a/drivers/soc/imx/sc/main/ipc.c
+++ b/drivers/soc/imx/sc/main/ipc.c
@@ -1,6 +1,6 @@
 /*
  * Copyright (C) 2016 Freescale Semiconductor, Inc.
- * Copyright 2017 NXP
+ * Copyright 2017-2018 NXP
  *
  * SPDX-License-Identifier:     GPL-2.0+
  */
@@ -321,10 +321,8 @@ int __init imx8_mu_init(void)
 		}
 
 		err = irq_set_irq_wake(irq, 1);
-		if (err) {
+		if (err)
 			pr_err("imx8mu_init: set_irq_wake failed: %d\n", err);
-			return err;
-		}
 	}
 
 	if (!scu_mu_init) {
-- 
1.7.9.5

